### DISCUSSION TITLE
kaggle notebook with LlamaCpp + OpenOrca-Zephyr-7B model

### DISCUSSION THREAD

**Discussion by wiiiktor** [guidance-ai/guidance#527] [Show and tell]
https://www.kaggle.com/wiiiktor/llamacpp-demo/edit

**Comment by wiiiktor** [comment]
Here you can find a Colab notebook, where I loaded a 4bit model, so it works even on a V100 gpu, which is always available (not like A100, which you can access from time to time), BUT I have a "UnsupportedOperation: fileno" message there; feedback on this would be welcome, so that I can fix and republish this notebook:

https://colab.research.google.com/drive/1D8VYz_Q1X7BSXj7vP326t_HKEbHnQ48_#scrollTo=ml8txfUavjY0

<img width="959" alt="image" src="https://github.com/guidance-ai/guidance/assets/41764319/7617576a-c0b3-46cf-bd52-38622a1c823c">

**Comment by Harsha-Nori** [comment]
@wiiiktor, probably a bit late to the update here, but this is due to a bug on the llamacpp-python side that we patched on our side (see this issue for details: https://github.com/guidance-ai/guidance/issues/545). Upgrading guidance should fix this issue :). Notebook looks great -- cool to see everything running on Colab!
